sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -1755,6 +1755,36 @@ __ESIMD_API simd<T, N> atomic_update(T *p, simd<unsigned, N> offset,
       p, offset, src1, src0, mask);
 }
 
+/// RAII-style class used to implement "semi-dynamic" SLM allocation.
```

Comment:
@petercaday, please review this part. E2E test - https://github.com/intel/llvm-test-suite/pull/1449.